sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1468,6 +1469,12 @@ object PushPredicateThroughJoin extends Rule[LogicalPlan] with PredicateHelper {
     case _ => false
   }
 
+  private def pushDownJoinConditions(conditions: Seq[Expression], plan: LogicalPlan) = {
+    conditions
+      .filterNot(_.semanticEquals(TrueLiteral)) // Push down true condition is useless.
```

Comment:
shouldn't this be optimized by `BooleanSimplification`? `true And cond` -> `cond`

Reply:
It is another issue. Will be fixed by another PR.